moonstone/normalization/reads/base.py

```diff
@@ -0,0 +1,33 @@
+class BaseDownsizing:
+
+    def __init__(self, raw_f, raw_r=None):
+        self.raw_f = raw_f
+        self.f = raw_f
+        self._downsized_f = None
+        if raw_r:
+            self.raw_r = raw_r
+            self.r = raw_r
+            self._downsized_r = None
+
+    def downsize_single(self):
+        """Overridden in child classes to perform specified downsizing of fragment reads"""
+        return self.f
+
+    def downsize_pair(self):
+        """Overridden in child classes to perform specified downsizing of paired-ends reads"""
+        return self.f, self.r
+
+    @property
+    def downsized_pair(self):
+        if getattr(self, "._downsized_f", None) is None:
+            self._downsized_f = self.downsize_pair()
```

Comment:
since the `downsize_pair` method returns a tuple, shouldn't it be something like: 

```python
self._downsized_f, self._downsized_r = self.downsize_pair()
```

Reply:
You're right, of course. I think that I wrote the downsize_single and then just copies it without adding the reverse.
This is fixed.
I'll push all these changes.